### ISSUE TITLE
Chat - Unread message green line marker remains after opening the new message

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#31623]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.1-7
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause-Internal Team
**Slack conversation:** @

## Action Performed:
1. Navigate to staging.new.expensify.com.
2. [User A] Create a group chat with User B and User C.
3. [User B] Open the group chat.
4. [User B] Navigate to a different chat.
5. [User B] Return to the group chat.

## Expected Result:
The unread message green line marker will disappear.

## Actual Result:
The unread message green line marker remains.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/85a4244e-8311-447d-a808-d8c0625d6e98

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
Not reproducible on production

https://github.com/Expensify/App/assets/115492554/eafa96b2-8c21-47c7-be0b-5516e70a73b5

**Comment by s-alves10** [comment]
https://github.com/Expensify/App/issues/30696#issuecomment-1791243687
Let's apply this changes.

We have several issues related to new marker
https://github.com/Expensify/App/pull/30485
https://github.com/Expensify/App/issues/27456
https://github.com/Expensify/App/pull/30485#issuecomment-1784859939

I think we need to refactor the logic of calculatiing new marker

cc @cubuspl42

**Comment by MonilBhavsar** [comment]
Might be because of https://github.com/Expensify/App/pull/25935
Me and @gedu can help with this

**Comment by MonilBhavsar** [comment]
I was also looking at that condition. Did you test other cases?
And are you able to prioritise this considering this is a blocker?

**Comment by MonilBhavsar** [comment]
I agree with root cause. Why we want to add in `!report.lastReadTime`?
```js
(((lastReadTimeRef.current || '') < report.lastReadTime) || !report.lastReadTime)) {
```

**Comment by MonilBhavsar** [comment]
😄 That was a question. If you have any valid reason or though behind that. We can keep it.

**Comment by MonilBhavsar** [comment]
@nkuoch if you don't mind, I would like to assign @tienifr to this issue

**Comment by MonilBhavsar** [comment]
> At the beginning i want to delete cacheUnreadMarkers even when report.lastReadTime is undefined, but after users visit the chat, report.lastReadTime has the value ('2023-11-21 14:57:37.414') and lastReadTimeRef.current is till undefined so deleting it at that time is good enough

Yes sounds good to me. @gedu if you have any thoughts on this

**Comment by luacmartins** [comment]
@tienifr are you able to work on this PR within the next hour?

**Comment by luacmartins** [comment]
Thanks!

**Comment by gedu** [comment]
sounds good to me

**Comment by Nathan-Mulugeta** [comment]
Hey team, I think this issue might have the same RCA with [this one](https://github.com/Expensify/App/issues/21453) or maybe it might be better to resolve that issue on the same PR as here. What are your thoughts?

**Comment by MonilBhavsar** [comment]
That's expected behavior. I commented in that issue

**Comment by anmurali** [comment]
I paid @tienifr and it doesn't look like there is an external issue reporter.
